rrmngmnt/foreman_host_wrapper.py

```diff
@@ -0,0 +1,301 @@
+        except Exception as ex:
+            raise ForemanException(
+                "Failed to remove element %s from foreman: %s" %
+                (element_name, ex)
+            )
+
+    def is_exist(self, element_name):
+        """
+        Check if element exist under foreman collection
+
+        :param element_name: element name
+        :type element_name: str
+        :return: True, if element exist, otherwise False
+        :rtype: bool
+        """
+        return bool(self.get_element_id(element_name=element_name))
+
+    def get_status(self, element_name):
+        """
+        Get element status
+
+        :param element_name: element name
+        :type element_name: str
+        :return: element status
+        :rtype: str
+        """
+        element_id = self.get_element_id(element_name=element_name)
+        element_status_d = self.collection.status(element_id)
+        if not element_status_d:
+            self.logger.debug(
+                "Failed to get element %s status" % element_name
+            )
+            return ""
+        return element_status_d["status"]
+
+
+class ForemanHost(Host):
+    """
+    Foreman host class
+    """
+    def __init__(
+        self, host_ip, host_root_password,
+        foreman_url, foreman_user, foreman_password
+    ):
+        """
+        Initialize foreman host class
+
+        :param host_ip: host ip
+        :type host_ip: str
+        :param host_root_password: host root password
+        :type host_root_password: str
+        :param foreman_url: foreman url
+        :type foreman_url: str
+        :param foreman_user: foreman user
+        :type foreman_user: str
+        :param foreman_password: foreman password
+        :type foreman_password: str
+        """
+        super(ForemanHost, self).__init__(ip=host_ip)
+        self.users.append(RootUser(password=host_root_password))
+        self.f_host_collection = ForemanCollection(
+            collection_name=FOREMAN_COLLECTION_HOST,
+            foreman_url=foreman_url,
+            foreman_user=foreman_user,
+            foreman_password=foreman_password,
+        )
+
+    def add_to_foreman(self, **kwargs):
+        """
+        Add host to foreman
+
+        :param kwargs: mac = str
+                       location_id = str
+                       domain_id = str
+                       organization_id = str
+                       medium_id = str
+                       architecture_id = str
+                       operatingsystem_id = str
+                       ptable_id = str
+                       hostgroup_id = str
+                       build = bool
+                       managed = bool
+        """
+        mac_address = kwargs.pop("mac", self.network.get_mac_by_ip(self.ip))
+        host_d = {
+            "name": self.fqdn,
+            "mac": mac_address,
+            "ip": self.ip,
+            "root_pass": crypt.crypt(self.root_user.password, "$1$redhat")
```

Comment:
it salt for encryption, because foreman not receive password with length less than 8 characters we need to encrypt our password and pass encrypted version to foreman